src/inmanta/util/dict_path.py

```diff
@@ -122,6 +122,15 @@ def escape(self) -> str:
     def matches(self, value: Optional[object]) -> bool:
         if value is None:
             return False
+
+        # Attempt numeric comparison if both values can be converted to floats
+        try:
+            if float(self._value) == float(value):
+                return True
+        except ValueError:
+            pass
+
+        # Fallback to string comparison
         return self._value == str(value)
```

Comment:
I think this is slightly more efficient when both values are float and different + it reads nicer
```suggestion
        # Attempt numeric comparison if both values can be converted to floats
        try:
            return float(self._value) == float(value):
        except ValueError:
            # Fallback to string comparison
            return self._value == str(value)

        
```

Reply:
indeed but I changed it :see_no_evil: